enterprise/admin/views.py

```diff
@@ -315,13 +315,29 @@ def _handle_singular(cls, enterprise_customer, manage_learners_form):
         """
         form_field_value = manage_learners_form.cleaned_data[ManageLearnersForm.Fields.EMAIL_OR_USERNAME]
         email = email_or_username__to__email(form_field_value)
+        existing_record = None
         try:
-            validate_email_to_link(email, form_field_value, ValidationMessages.INVALID_EMAIL_OR_USERNAME, True)
+            existing_record = validate_email_to_link(email, form_field_value, ValidationMessages.
+                                                     INVALID_EMAIL_OR_USERNAME, True)
         except ValidationError as exc:
             manage_learners_form.add_error(ManageLearnersForm.Fields.EMAIL_OR_USERNAME, exc)
-        else:
-            EnterpriseCustomerUser.objects.link_user(enterprise_customer, email)
-            return [email]
+        if existing_record:
```

Comment:
The else: portion of the try-except should not be removed. Otherwise, if a ValidationError is raised, we will still try to call link_user() with the invalid email address.

Reply:
Good catch. Thank you for pointing this out!